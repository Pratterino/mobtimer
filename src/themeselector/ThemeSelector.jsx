import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import ThemeButton from "./../themebutton/ThemeButton";
import "./ThemeSelector.scss";

function ThemeSelector(props) {
    const [tempTheme, setTempTheme] = useState(props.settings.theme);
    const themes = ["ladies-night-theme", "sublime-theme"];

    useEffect(() => {
        window.document.body.className = tempTheme || props.settings.theme;
    }, [props.settings.theme, tempTheme]);

    return (
        <div className="theme-selector">
            {themes.map(theme =>
                <ThemeButton
                    theme={theme}
                    key={theme}
                    setTempTheme={setTempTheme}
                    className="theme-selector__button"
                />
            )}
        </div>
    );
}

const mapStateToProps = state => ({
    settings: state.settings,
});

const mapDispatchToProps = dispatch => bindActionCreators({
}, dispatch);

ThemeSelector.propTypes = {
    setTheme: PropTypes.func.isRequired,
};

export const unwrapped = ThemeSelector;
export default connect(mapStateToProps, mapDispatchToProps)(ThemeSelector);
