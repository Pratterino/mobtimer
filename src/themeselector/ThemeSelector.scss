.theme-selector {
  .theme-selector__button {
  }
}
