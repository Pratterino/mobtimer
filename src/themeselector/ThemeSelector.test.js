import React from 'react';
import {unwrapped as UnwrappedThemeSelector} from './ThemeSelector';
import {shallow} from "enzyme";

describe('ThemeSelector', () => {
    let wrapper;

    const renderComponent = (props) => (
        wrapper = shallow(
            <UnwrappedThemeSelector
                settings={{
                    theme: "ducca-evening",
                }}
                {...props}
            />
        )
    );

    beforeEach(() => {
        renderComponent();
    });

    describe('render', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    })
});
