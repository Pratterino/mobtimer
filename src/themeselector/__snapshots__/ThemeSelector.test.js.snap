// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSelector render should match snapshot 1`] = `
<div
  className="theme-selector"
>
  <Connect(ThemeButton)
    className="theme-selector__button"
    key="ladies-night-theme"
    setTempTheme={[Function]}
    setTheme={[Function]}
    theme="ladies-night-theme"
  />
  <Connect(ThemeButton)
    className="theme-selector__button"
    key="sublime-theme"
    setTempTheme={[Function]}
    setTheme={[Function]}
    theme="sublime-theme"
  />
</div>
`;
